src/main/java/com/example/animalshelter/service/AnimalNotFoundException.java

```diff
@@ -0,0 +1,20 @@
+package com.example.animalshelter.service;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+/**
+ * Throw this exception when delete action failed.
+ * Like when animal id is not present id database.
+ */
+@ResponseStatus(HttpStatus.NOT_FOUND)
+public class AnimalNotFoundException extends Exception {
+
+    public AnimalNotFoundException(String message) {
+        super(message);
+    }
+
+    public AnimalNotFoundException() {
+        super();
+    }
```

Comment:
I would move the line
`private final static String ANIMAL_NOT_FOUND_MSG = "Animal with provided id doesn't exist in the database.";`
from `AnimalRepositoryListImpl.java` here and pass id to constructor to have it in the message, so sth like that:
```suggestion
    public AnimalNotFoundException(Integer id) {
        super(String.format("Animal with id %d doesn't exist in the database."), id);
    }
```

Reply:
done